pkcs7/src/CryptographicMessageSyntax2004.rs

```diff
@@ -0,0 +1,534 @@
+//! Selected structures from RFC5652
+
+use x509::{AttributeTypeAndValue, Certificate, Name, SubjectKeyIdentifier};
```

Comment:
I suspect that we don't want `pkcs*` crates to depend on `x509`. We probably need to refactor some of the types in that crate into a shared crate. For example, `AttributeTypeAndValue` and `Name` are likely to be needed in a variety of places. The same with x509 extensions (not relevant to this commit). This probably requires some discussion.